docs/do-not-merge.md

```diff
@@ -0,0 +1,815 @@
+
+[#14407](https://github.com/apache/druid/pull/14407)
+
+### Streaming ingestion with concurrent replace
+
+Streaming jobs reading from Kafka and Kinesis with `APPEND` locks can now ingest concurrently with compaction running with `REPLACE` locks. The segment granularity of the streaming job must be equal to or finer than that of the concurrent replace job.
+
+[#15039](https://github.com/apache/druid/pull/15039)
+
+## Ingestion
+
+### JSON and auto column indexer
+
+The `json` column type is now equivalent to using `auto` in JSON-based batch ingestion dimension specs. Upgrade your ingestion specs to `json` to take advantage of the features and functionality of `auto`, including the following:
+
+- Type specializations including ARRAY typed columns
+- Better support for nested arrays of strings, longs, and doubles
+- Smarter index utilization
+
+`json` type columns created with Druid 28.0.0 are not backwards compatible with Druid versions older than 26.0.0.
+If you upgrade from one of these versions, you can continue to write nested columns in a backwards compatible format (version 4).
+
+For more information, see [Nested column format in the upgrade notes](#nested-column-format).
+
+[#14955](https://github.com/apache/druid/pull/14955)
+[#14456](https://github.com/apache/druid/pull/14456)
+
+### Ingestion status
+
+Ingestion reports now include a `segmentLoadStatus` object that provides information related to the ingestion, such as duration and total segments.
+
+[#14322](https://github.com/apache/druid/pull/14322)
+
+### SQL-based ingestion
+
+#### Ability to ingest ARRAY types
+
+SQL-based ingestion now supports storing ARRAY typed values in [ARRAY typed columns](https://druid.apache.org/docs/latest/querying/arrays) as well as storing both VARCHAR and numeric typed arrays.
+Previously, the MSQ task engine stored ARRAY typed values as [multi-value dimensions](https://druid.apache.org/docs/latest/querying/multi-value-dimensions) instead of ARRAY typed columns.
+
+The MSQ task engine now includes the `arrayIngestMode` query context parameter, which controls how
+`ARRAY` types are stored in Druid segments.
+Set the `arrayIngestMode` query context parameter to `array` to ingest ARRAY types.
+
+In Druid 28.0.0, the default mode for `arrayIngestMode` is `mvd` for backwards compatibility, which only supports VARCHAR typed arrays and stores them as multi-value dimensions. This default is subject to change in future releases.
+
+Note that this improvement is incompatible with previous Druid versions. For information on how to migrate to the new behavior, see the [Ingestion options for ARRAY typed columns in the upgrade notes](#ingestion-options-for-array-typed-columns).
```

Comment:
this isn't strictly true, the ARRAY columns were introduced in Druid 27, the only problem would be if someone was trying to use the new `"arrayIngestMode":"array"` and had not upgraded all of the nodes the task run on resulting in a mix of behavior

Reply:
Removed the sentence about incompatibility with previous Druid versions.